src/disk_buffer_pool.cpp

```diff
@@ -219,21 +219,22 @@ namespace libtorrent
 
 // this function allocates buffers and
 // fills in the iovec array with the buffers
-	int disk_buffer_pool::allocate_iovec(file::iovec_t* iov, int iov_len)
+	int disk_buffer_pool::allocate_iovec(span<file::iovec_t> iov)
 	{
 		std::unique_lock<std::mutex> l(m_pool_mutex);
-		for (int i = 0; i < iov_len; ++i)
+		for (auto& i : iov)
 		{
-			iov[i].iov_base = allocate_buffer_impl(l, "pending read");
-			iov[i].iov_len = block_size();
-			if (iov[i].iov_base == nullptr)
+			i.iov_base = allocate_buffer_impl(l, "pending read");
+			i.iov_len = block_size();
+			if (i.iov_base == nullptr)
 			{
 				// uh oh. We failed to allocate the buffer!
 				// we need to roll back and free all the buffers
 				// we've already allocated
-				for (int j = 0; j < i; ++j)
+				for (auto j : iov)
```

Comment:
I know that probably the extra cycles will have zero impact, but I find more explicit the algorithm idea as it was, even if it's trivial. Just an opinion.